fv3core/stencils/c_sw.py

```diff
@@ -116,6 +140,25 @@ def transportdelp(
         ptc = (pt * delp + (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea) / delpc
         wc = (w * delp + (fx2 - fx2[1, 0, 0] + fy2 - fy2[0, 1, 0]) * rarea) / delpc
 
+    with computation(PARALLEL), interval(...):
+        # update vorticity and kinetic energy
+        compile_assert(grid_type < 3)
```

Comment:
We've been putting these in the init method of the objects by our current convention... But might be worthwhile to leave this one here near where the code would branch, given this object is quite large.